content/cloud-docs/agents/hooks.mdx

```diff
@@ -0,0 +1,157 @@
+---
+page_title: Hooks - Terraform Cloud Agents - Terraform Cloud and Terraform Enterprise
+description: >-
+  Terraform Cloud Agent hooks are custom programs that run at strategic points
+  during Terraform runs.
+---
+
+# Hooks
+
+Terraform Cloud Agents support running custom programs, or hooks, during
+strategic points of a Terraform run. These hooks allow you to extend the
+functionality of Terraform runs. For example, you may create a hook to
+dynamically download software required by the Terraform run or send an HTTP
+request to a system to kick off an external workflow.
+
+## Supported Hooks
+
+The following hooks are supported:
+
+- `pre-plan` - The hook runs before `terraform init` during a `plan` operation.
+
+- `post-plan` - This hook runs after `terraform plan` during a `plan`
+operation regardless whether or not `terraform plan` completed successfully.
+
+- `pre-apply` - The hook runs before `terraform init` during an `apply`
+operation.
+
+- `post-apply` - This hook runs after `terraform apply` during an `apply`
+operation regardless whether or not `terraform apply` completed successfully.
+
+## Hook Behavior 
+
+Please note the following behavior when using hooks:
+
+- If a hook exits with a non-zero exit code, the Terraform run will fail
+immediately.
+- The standard output and standard error from the hook will be printed
+alongside the Terraform run output.
+- The name of each hook is not customizable.
```

Comment:
```suggestion
-  The hook name must match one of the [supported types](#supported-types). You cannot customize or change these names. Because of this, there can only be one hook of each type configured for each agent. For example, you could create a `pre-plan` and `pre-apply` hook, but you cannot create two `pre-plan` hooks.
```